ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1330,6 +1332,9 @@ void close() throws StoreException {
     long startTimeInMs = time.milliseconds();
     try {
       persistor.write();
+      if (persistorTask != null) {
```

Comment:
Good idea! Changes are made.

Reply:
The `persistorTask.cancel(false)` allows any in progress task to complete background. However, after that, we invoke `persistor.write()` to flush persistor again. Note that `write()` is an synchronized method, so it will be blocked until any background task completes. This guarantees persistor is canceled before `close` method completes.